tests/MongoDB.Driver.Core.Tests/Core/Configuration/ConnectionStringTests.cs

```diff
@@ -110,6 +110,49 @@ public void Resolve_with_resolveHosts_should_return_expected_result(string conne
             result.ToString().Should().Be(expectedResult);
         }
 
+        [Theory]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tlsInsecure=true&tlsInsecure=false", true)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tlsInsecure=false&tlsInsecure=true", true)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tlsInsecure=true&tlsInsecure=true", false)]
+        public void With_more_then_one_tlsInsecure(string connectionString, bool shouldThrow)
+        {
+            if (shouldThrow)
+            {
+                var ex = Record.Exception(() => { var _ = new ConnectionString(connectionString); });
+
+                var e = ex.Should().BeOfType<MongoConfigurationException>().Subject;
+                e.Message.Should().Be("tlsInsecure has already been configured with a different value.");
+            }
+            else
+            {
+                var _ = new ConnectionString(connectionString);
+            }
+        }
+
+        [Theory]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tls=true&tls=false", true)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?ssl=false&ssl=true", true)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?ssl=false&tls=true", true)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tls=false&ssl=true", true)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?ssl=false&ssl=false", false)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tls=false&tls=false", false)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?ssl=true&ssl=true", false)]
+        [InlineData("mongodb://test5.test.build.10gen.cc?tls=true&tls=true", false)]
+        public void With_more_then_one_tls_or_ssl(string connectionString, bool shouldThrow)
+        {
+            if (shouldThrow)
+            {
+                var ex = Record.Exception(() => { var _ = new ConnectionString(connectionString); });
+
+                var e = ex.Should().BeOfType<MongoConfigurationException>().Subject;
+                e.Message.Should().Be("tls has already been configured with a different value.");
+            }
+            else
+            {
+                var _ = new ConnectionString(connectionString);
+            }
```

Comment:
Same comment as above:

    var exception = Record.Exception(() => { var _ = new ConnectionString(connectionString); });

    if (shouldThrow)
    {
        var e = exception.Should().BeOfType<MongoConfigurationException>().Subject;
        e.Message.Should().Be("tls has already been configured with a different value.");
    }
    else
    {
        exception.Should().BeNull();
    }